### ISSUE TITLE
Issue with detection of English text

### ISSUE THREAD

**Issue by solution7** [patrickschur/language-detection#17]
> Phrase : A beautiful villa in eastern Sweden --- Language Detected : sv

Phrase : Uma bela casa no leste da Suécia --- Language Detected : pt-BR
Phrase : Eine schöne Villa in Ost-Schweden --- Language Detected : de
Phrase : En vacker villa i östra sverige --- Language Detected : sv
Phrase : পূর্ব সুইডেনে একটি সুন্দর বাগানবাড়ি --- Language Detected : bn
Phrase : Une belle villa en Suède orientale --- Language Detected : it
Phrase : فيلا جميلة في شرق السويد --- Language Detected : ar
Phrase : En vakker villa i Øst-Sverige --- Language Detected : sv
Phrase : Una hermosa villa en el este de Suecia --- Language Detected : es
Phrase : En smuk villa i det østlige Sverige --- Language Detected : dk
Phrase : Kaunis huvila Itä Ruotsissa --- Language Detected : fi
Phrase : पूर्वी स्वीडन में एक खूबसूरत विला --- Language Detected : hi
Phrase : Una bella villa in Svezia orientale --- Language Detected : it
Phrase : 東部スウェーデンの美しいヴィラ --- Language Detected : ja
Phrase : O vilă frumoasă în estul Suediei --- Language Detected : ro
Phrase : Isang magandang villa sa silangang Sweden --- Language Detected : jv
Phrase : Красивая вилла в восточной части Швеции --- Language Detected : ru
Phrase : Красива вілла в східній частині Швеції --- Language Detected : uk
Phrase : A beautiful villa in eastern Sweden --- Language Detected : sv
Phrase : Uma bela casa no leste da Suécia --- Language Detected : pt-BR
Phrase : Eine schöne Villa in Ost-Schweden --- Language Detected : de
Phrase : En vacker villa i östra sverige --- Language Detected : sv
Phrase : পূর্ব সুইডেনে একটি সুন্দর বাগানবাড়ি --- Language Detected : bn
Phrase : Une belle villa en Suède orientale --- Language Detected : it
Phrase : فيلا جميلة في شرق السويد --- Language Detected : ar
Phrase : En vakker villa i Øst-Sverige --- Language Detected : sv
Phrase : Una hermosa villa en el este de Suecia --- Language Detected : es
Phrase : En smuk villa i det østlige Sverige --- Language Detected : dk
Phrase : Kaunis huvila Itä Ruotsissa --- Language Detected : fi
Phrase : पूर्वी स्वीडन में एक खूबसूरत विला --- Language Detected : hi
Phrase : Una bella villa in Svezia orientale --- Language Detected : it
Phrase : 東部スウェーデンの美しいヴィラ --- Language Detected : ja
Phrase : O vilă frumoasă în estul Suediei --- Language Detected : ro
Phrase : Isang magandang villa sa silangang Sweden --- Language Detected : jv
Phrase : Красивая вилла в восточной части Швеции --- Language Detected : ru
Phrase : Красива вілла в східній частині Швеції --- Language Detected : uk
Phrase : A beautiful villa in eastern Sweden --- Language Detected : sv
Phrase : Uma bela casa no leste da Suécia --- Language Detected : pt-BR
Phrase : Eine schöne Villa in Ost-Schweden --- Language Detected : de
Phrase : En vacker villa i östra sverige --- Language Detected : sv
Phrase : পূর্ব সুইডেনে একটি সুন্দর বাগানবাড়ি --- Language Detected : bn
Phrase : Une belle villa en Suède orientale --- Language Detected : it
Phrase : فيلا جميلة في شرق السويد --- Language Detected : ar
Phrase : En vakker villa i Øst-Sverige --- Language Detected : sv
Phrase : Una hermosa villa en el este de Suecia --- Language Detected : es
Phrase : En smuk villa i det østlige Sverige --- Language Detected : dk
Phrase : Kaunis huvila Itä Ruotsissa --- Language Detected : fi
Phrase : पूर्वी स्वीडन में एक खूबसूरत विला --- Language Detected : hi
Phrase : Una bella villa in Svezia orientale --- Language Detected : it
Phrase : 東部スウェーデンの美しいヴィラ --- Language Detected : ja
Phrase : O vilă frumoasă în estul Suediei --- Language Detected : ro
Phrase : Isang magandang villa sa silangang Sweden --- Language Detected : jv
Phrase : Красивая вилла в восточной части Швеции --- Language Detected : ru
Phrase : Красива вілла в східній частині Швеції --- Language Detected : uk

Above are the phrase and the language detected. The library seems to be working fine, but its not detecting the correct for simple english text.

**Comment by patrickschur** [comment]
You have to know that english is very similiar to other languages and the sentences you choosed are not very long, this two things make it very hard to detect the right language. On the other side this library is more or less optimized on speed. If you want to improve the detection phase please have a look at [here](https://github.com/patrickschur/language-detection#how-can-i-improve-the-detection-phase). Where I described the procedure how to improve the detection.

Maybe I should increase the detection by default a little bit?

If you need any further help let me know. :)

**Comment by nachitox** [comment]
What about setting maxLength? Does a bigger number improve detection?

**Comment by patrickschur** [comment]
The length of ngrams does not improve the detection much. Maybe setting the maxLength to 4 or 5 will make a difference but I wouldn't go further otherwise it's getting even worse.

**Comment by iquito** [comment]
I have used a maxLength of 4, it helped my detection results quite a bit (together with setting maxNgrams to 9000).

The library also "only" includes the Universal Declaration of Human Rights as the base for the language detection, so many common phrases and words will be missing (like country names, idioms, greetings, specific nouns). Depending on the usage it makes sense to add additional words/sentences to the language samples, so they can factor in. I did it for my use case (analyzing emails) and it worked very well.
